test/vmtests/vmtests/utils/libvirt_utils.py

```diff
@@ -136,21 +252,49 @@ def _add_disk_xml(
     disk_source = ET.SubElement(disk, "source")
     disk_source.attrib["file"] = file_path
 
+# Adds a disk to a libvirt domain XML document.
+def _add_iso_xml(
```

Comment:
If you add a `readonly` param to `_add_disk_xml`, you could implement `_add_iso_xml` using `_add_disk_xml`.